tart2ms/tart2ms.py

```diff
@@ -756,41 +948,53 @@ def __twelveball(direction):
             'STATE_ID': (("row",), da.from_array(np.zeros(row, dtype=int), chunks=chunks['row'])),
         }
-        if fill_model and sources:
+        if fill_model:
             main_table["MODEL_DATA"] = (dims, model_data)
         dataset = Dataset(main_table)
         ms_datasets.append(dataset)
 
     ms_writes = xds_to_table(ms_datasets, ms_table_name, columns="ALL")
+    # auxilary table futures creation
     spw_writes = xds_to_table(spw_datasets, spw_table_name, columns="ALL")
     ddid_writes = xds_to_table(ddid_datasets, ddid_table_name, columns="ALL")
+    for tt in [ant_table, feed_table, field_table, pol_table, obs_table, src_table]:
+        tt.write()
 
-    dask.compute(ms_writes)
-
-    ant_table.write()
-    feed_table.write()
-    field_table.write()
-    pol_table.write()
-    obs_table.write()
-    src_table.write()
-
-    dask.compute(spw_writes)
-    dask.compute(ddid_writes)
+    # execute graph with futures
+    LOGGER.info("Synthesizing MS...")
+    with ProgressBar():
+        dask.compute([ms_writes + spw_writes + ddid_writes] + MSTable.get_futures())
 
+    LOGGER.info("Performing finalization of UVW coordinates if needed")
     if uvw_generator == 'telescope_snapshot':
-        pass
+        pass # user has been warned about their choice -- this cannot be used when rephasing
     elif uvw_generator == 'casacore':
         # rephasing requires us to tilt w towards the rephased point on the sphere
         # this is also needed if we haven't generated zenithal points yet because we didn't predict
         # a model
-        if phase_center_policy == 'rephase-obs-midpoint' or \
-           phase_center_policy == 'rephase-NCP' or \
-           phase_center_policy == 'rephase-SCP' or \
-           not fill_model:
-            fixms(ms_table_name)
+        if isinstance(phase_center_policy, str):
+            fn = phase_center_policy.replace("rephase-","")
+            fsrc = list(filter(lambda x: x['name'].upper() == fn and
+                                        x['position']["FRAME"] == "Special Body",
+                            read_known_phasings()))
+            is_special_body = len(fsrc) > 0
+        elif isinstance(phase_center_policy, SkyCoord):
+            is_special_body = False
+        else:
+            raise ValueError("Invalid type for phase_centre_policy - expect string or SkyCoord")
+        single_field = isinstance(phase_center_policy, SkyCoord) or \
+                       phase_center_policy.find('rephase-') == 0 or \
+                       phase_center_policy == "no-rephase-obs-midpoint"      
+        if not is_special_body: # non-sidereal tracking fields have their UVW computed per timestamp due to change in RA,DEC
+            if (single_field or # non-zeniths for which UVW are needed
+                (not single_field and not fill_model)): # zeniths not yet computed - no model predicts happened
+                fixms(ms_table_name)
```

Comment:
The only cases requiring fixvis to run is the zenethal multifield case where we didn't predict before requiring UVW and the single field cases where W should point towards the new field centre